### ISSUE TITLE
Aerie and Mule mass

### ISSUE THREAD

**Issue by accursedCursive** [endless-sky/endless-sky#3541]
Lionheart industries has a tendency to build with lightweight materials.
But the masses of these two ships of theirs seem ridiculous:

Aerie: At 130 mass, this is a carrier that weighs as much as an interceptor. It should be pointed out that this is barely heavier than a Headhunter (120 mass), which is a much smaller ship that is built with lightweight materials. Even if Aerie is built with expensive lightweight materials (it doesn't say), it should be about 300 mass.
Mule: A Bactrian with slightly smaller size and outfit space, but only a third of the mass, making it unusually light for a medium warship. Which is odd, because it looks the heaviest, and it has a similar appearance to a Bactrian which is very heavy despite being made by the same manufacturer.

**Comment by Nescio0** [comment]
Let's have some things clear first:
hull mass = `mass`
empty mass = `mass` + `outfit space`
effective mass = `mass` + `outfit space` + `cargo space`

Also, increasing a ship's mass means increasing its heat capacity as well.

Now let's make a comparison of Lionheart ships by increasing mass:
```
             mass = hull + outfits ; cargo ; bunks ; fuel
Dagger     :  110 =   20 +      90 ;    -- ;     1 ;   --
Flivver    :  170 =   40 +     130 ;     5 ;     3 ;  500
Scout      :  330 =  110 +     220 ;    55 ;    12 ;  800
Headhunter :  370 =  120 +     250 ;    50 ;     4 ;  400
Raven      :  410 =  130 +     280 ;    30 ;    13 ;  500
Aerie      :  520 =  130 +     390 ;    50 ;    28 ;  500
Corvette   :  570 =  150 +     420 ;    40 ;    32 ;  500
Mule       :  770 =  320 +     450 ;   270 ;    43 ;  600
Bactrian   : 1680 =  940 +     740 ;   530 ;   245 ;  700
```

As you can see the Aerie is clearly heavier than any interceptor, and the Bactrian is more than twice as heavy as the Mule. While I do not have any objections to increasing ships' masses, I do not think it's really necessary in this case.

**Comment by jafdy** [comment]
When loaded down with fighters and outfits the Aerie is no longer a light ship.

**Comment by KiLEdEnNis** [comment]
imagine; the Aerie and Mule have 2 and 1 fighter bays respectively, realistically that means that they have hangars built-in; big empty areas where the fighters would dock
so the Aerie has 2 decent empty areas in it for the fighters
as for the Mule, imagine a Z-axis as well; the Bactrian is logically a bit taller than the Mule as well, and it's generally quite a bit larger anyway, which means their masses are realistic compared to each other, even though the Bactrian has 3 fighter bays

furthermore, changing balance for the sake of a tiny detail in "lore" is a bad decision as well

**Comment by accursedCursive** [comment]
Even diluted by outfits, the mass difference gives Aerie a huge advantage in mobility-based combat.
Bastion is a comparable ship to Aerie, and assuming both ships have full outfits and no cargo...
Bastion: 580 mass, 470 outfit space, 1050 mass and 0.44 outfit space per mass unit
Aerie: 130 mass, 390 outfit space, 520 mass and 0.75 outfit space per mass unit
I think that Aerie should be fairly lightweight, but this is just ridiculous.
Even the Headhunter, a light warship that's expensive due to its light weight, only gets 0.68 outfit space per mass unit.
Having a high outfit space per mass unit means you can be very powerful yet also very agile.
For this reason the Aeries in my game decided they wanted to operate as interceptor hunters, following interceptors into deep space and lasering them to death, even though it's meant to be Ravens doing stuff like that.
So I doubled their mass, same for Corvette.

**Comment by KiLEdEnNis** [comment]
comparing the Aerie to the Bastion?
the Aerie uses light materials and weighs 520, read back for my fighter bay thing: 2 stock Daggers bring its mass up to 738
the Bastion has 2 more guns, 1k more shields, 2,5k more hull, 80 more outfit space, 30 more weapon space, 25 more engine space
as in one can't compare the Aerie & Bastion since the Bastion is generally a heavier warship than the Aerie

speaking about outfit space per mass unit? again: the Aerie has 2 empty fighter bays, which saves a lot of mass since 2 109 mass Daggers can be docked in them, meaning the base mass of the Aerie is technically 348 w/ the Daggers docked, bringing the outfit space per mass to 0.47 (348/738), not that an outfit space to mass ratio is really important

a short summary: the Aerie has 2 fighter bays which means empty space IN the ship hull, thus it naturally weighs less

as for the Corvette, it weighs 150 by default; it's an expensive lightweight ship
similar ships' mass:
Splinter: 250
Frigate: 310
Osprey: 270
as in the Corvette is fine too, and your arguments are, surprisingly, wrong

**Comment by KiLEdEnNis** [comment]
> Fighters aren't outfits though.

you're missing the point; the ship hulls have big empty bays for the fighters to be in, take that away and the ship itself will be quite a bit smaller
if the Aerie didn't have fighter bays it'd nearly be a light warship

either way your changes are unnecessary, if ya want that for yourself go ahead though

**Comment by accursedCursive** [comment]
I understand that the Aerie has fighter bays.
But what I'm saying is that the Aerie is beyond lightweight.
The Headhunter is lightweight, and Aerie makes it seem heavy by comparison.

**Comment by Nescio0** [comment]
Seriously, outfits/mass? Mass + outfits is what matters. I personally prefer massive ships, since total mass increases heat capacity, does not affect maximum speed, and only slightly lowers accelaration.

The Aerie already is 40% heavier than the Headhunter, so I don't think a mass increase is necessary here. If anything, I think the Bastion and Hawk deserve a mass reduction, e.g. –60 mass; the former because its total mass (1050) is very close to significantly better ships such as the Vanguard (1060), Falcon (1050), and Leviathan (1100); the latter because its total mass (350) is close to many light warships while it actually is hardly better than the Fury (230). However, it probably isn't important enough for a pull request.

**Comment by KiLEdEnNis** [comment]
the Aerie is very lightweight, but not beyond it; if it has 2 Daggers docked its mass would be 348, more than even a Frigate
imagine nearly the whole side area of the Aerie are those hangars, only the long base would be the ship itself, and that's hardly more than the Headhunter, so the mass comparison is fine
increasing the Aerie mass just because it's too light even though it's very heavy when it has fighters docked is a no, and I'll refer back to `changing balance for the sake of a tiny detail in "lore" is a bad decision as well`

**Comment by jafdy** [comment]
Has anyone played with these changes yet?

**Comment by accursedCursive** [comment]
@jafdy I've played with an extra 70 mass for Aerie and Corvette, and it's an improvement.
Aerie is no longer hunting down interceptors, and Corvette is no longer overpowered as a bomber.

@KiLEdEnNis Mass with fighters docked doesn't matter in combat.
Carriers will _always_ deploy their fighters in combat.
The result is that Aerie is too light in combat.

**Comment by Tadrix** [comment]
> Carriers will _always_ deploy their fighters in combat.

Incorrect. It could change very soon. #3383 #3441

**Comment by tehhowch** [comment]
Even on the current `master`, if those fighters are badly injured they will return to the carrier.

**Comment by accursedCursive** [comment]
That's nice, but Aerie, Corvette, and Mule are still underweight.
The percentage of a ship's mass that is outfits depends on what that ship is meant to be: >70% is for the lightest of the light, 60-70% is popular for interceptors and light warships, ~50% is typical of medium warships and anything below is probably a cargo ship.
The three aforementioned ships are 75%, 73%, and 58% respectively, both way above their expected ratios.

**Comment by tehhowch** [comment]
>The percentage of a ship's mass that is outfits depends on what that ship is meant to be

I am unaware of this being a recommendation made when guiding content creators (it is not listed in the CreatingShips wiki). Is this an observation? If so, you should provide a nice breakdown of the current content (probably a scatter plot color-coded by manufacturer so we can observe the specific trends separate from the trends that go with each ship tier, with labels for each of the game's categories).

**Comment by KiLEdEnNis** [comment]
@Tadrix I put my PR on hold due to college & shit, I'm not sure when I'll have time to finish it, hopefully within a week (test week this sweek means short college days & next week probably no college)
but I'm new to C++ so I'm mostly relying on looking C++ up & asking Tehhowch

**Comment by KiLEdEnNis** [comment]
> The percentage of a ship's mass that is outfits depends on what that ship is meant to be: >70% is for the lightest of the light, 60-70% is popular for interceptors and light warships, ~50% is typical of medium warships and anything below is probably a cargo ship.

[citation needed]

**Comment by endless-sky** [comment]
You seem to be approaching this from exactly the opposite direction I usually do when designing a ship.

My approach is to set the speed penalty values of the chassis (drag and mass) to whatever values, within reason, give the fully-loaded ship the sort of performance that I think it should have. In the case of the Aerie, "fully-loaded" includes two fighters, which adds a *lot* of mass.

Remember that this is a game, and that interesting and fun game-play is more important than realism.

**Comment by KiLEdEnNis** [comment]
> Remember that this is a game, and that interesting and fun game-play is more important than realism.

(adding that the low mass is not even that unrealistic to begin with)

**Comment by accursedCursive** [comment]
@endless-sky But you forgot to consider that those fighters aren't going to be attached in combat (ignoring new NPC behaviours, which the Aerie long precedes).
Which means the Aerie is more agile than you intended.
You can't see it, because vanilla ES has super weak engines and "sit still and hold tab for 10 seconds" style combat so a slight mass difference makes almost zero difference to speed and combat effectiveness.
But I can see that the Aerie is disproportionately agile, because I made thrusters 6x more powerful and it zooms around eating interceptors without a mass increase.
The drag is fine, it can't catch an interceptor in a straight-line chase, but in any other situation an unnerfed Aerie in is the worst enemy of small ships.
Except for maybe Leviathans, but that's only because of what I made their default loadout.

**Comment by Amazinite** [comment]
>You can't see it, because vanilla ES has super weak engines and "sit still and hold tab for 10 seconds" style combat so a slight mass difference makes almost zero difference to speed and combat effectiveness.
But I can see that the Aerie is disproportionately agile, because I made thrusters 6x more powerful and it zooms around eating interceptors without a mass increase.

So the only reason you've raised this issue is because you've modded the game, and that mod has lead the Aerie to being too agile, but **it can not be noticed in the vanilla game**? My response to that: don't suggest changes to the vanilla game **based off of your modded version of the game**, and change the Aerie mass all you want in the modded version of the game. 

Engine space and mass affect acceleration and steering. Acceleration is thrust/mass and acceleration is steering/mass. An abstraction of thrust and steering is simply engine space. Therefore, `engine space/mass` gives us a good idea of how agile a ship is (how comparatively high its steering rate and acceleration could be).
Let's assume that our Aerie is maxed out on outfits, empty on cargo, and has no fighters docked, then compare the agility of that ship to other ships. Here is the Aerie and the three human ships above and below it in terms of agility as described above.

Ship | Category | Agility 
------|-----|-----
Arrow | Transport | 0.194
Blackbird | Transport | 0.193
Manta | Medium Warship | 0.192
Aerie | Medium Warship | 0.183
Osprey | Medium Warship | 0.181
Corvette | Medium Warship | 0.175
Splinter | Medium Warship | 0.154

As you can see, the Aerie is practically just as agile as the Osprey, and still gets beat out by the Manta, and even still beats out the Corvette by this metric.

tl;dr It's fine.

**Comment by KiLEdEnNis** [comment]
> @endless-sky But you forgot to consider that those fighters aren't going to be attached in combat 

no, he didn't
realistically those carriers have empty hangars in their hull, which means they're lighter than other ships unless the fighters are docked
when fighters are deployed those carriers should be lighter than the average ship of their size because they are partially empty again
realistically this also means that carriers are generally less effective in combat than other ships of their size since they have less room in their hull to use

> Which means the Aerie is more agile than you intended.

man if you paid me a dollar for every time you made any assumption, I could employ a full professional team to make Endless Sky 3D

> But I can see that the Aerie is disproportionately agile, because I made thrusters 6x more powerful and it zooms around eating interceptors without a mass increase.

just... end my suffering, "the Aerie is too fast because my improperly balanced mod has engines that makes it that fast"
do you have any idea how ridiculously stupid and unprofessional you sound everytime you say something like this? the game and its content is balanced around itself, if you create trash overpowered content that's not our fault, stop blaming the base game for your own ignorance

**Comment by Startome** [comment]
Yeah github exists for recommendations to the base game. If you aren't even playing base game, stop suggesting anything until you are. Gracious.

That's like changing the motor and transmission in your car then filing a lawsuit against the manufacture when the transmission fails and you get hurt. Nobody else is responsible for changes you make to working products except yourself.

**Comment by Tadrix** [comment]
> you forgot to consider that those fighters aren't going to be attached in combat

You seem to forget that fighters are attached to carriers always, except when in combat.

Consider a stereotypical slow carrier. If you make it to be slow _without_ its fighters, then with them this carrier would be painfully slow, which would increase frustration of players. Hence in _Endless Sky_ starships are balanced not around hull weight (which would mean no means to defend, power itself, or, you know, fly), but around curb weight (which means a vehicle is functional). Since one of main functions of carriers involve fighter operations, this includes docked fighters (duh).

**Comment by accursedCursive** [comment]
@KiLEdEnNis 
What do you mean, "improperly balanced mod"?
It's way better balanced than the vanilla game.
The vanilla game has to make hulls get exponentially more expensive with size because the balance favours large ships, so you end up with this situation where:
- Small ships are overpowered relative to their cost
- Large ships are overpowered relative to their mass
- There's no reason not to use the most expensive outfits on large ships, because their price pales in comparison to the hull price
- Because a ship half the size is a quarter of the effectiveness, or something like that, there's no reason to fly anything but the largest ship in your fleet

In my game:
- Small ships are overpowered relative to their mass, which means flying one in a fleet containing larger ships is a viable option
- Ships are balanced relative to their cost, 10mil worth of interceptors are as useful as 10mil worth of medium warships
- Equipping any ship with really expensive outfits will make the ship double the regular cost

An interesting phenomenon is that as a rule of thumb, small ships will beat an equal value of large ships, large ships will beat medium ships, and medium ships will beat small ships.
In contrast, the vanilla game is just small-beats-large, because larger ships had the advantage of being least likely to die (in my game, odds of survival are more balanced between sizes).
Of course, my rule of thumb totally falls apart outside of human space, because different races are balanced differently: Such as the Quarg, who use large ships for support and rely almost entirely on medium and small ships for engagements, or the Korath where the pecking circle goes the other way around due to their extreme lethality:survivability ratio.

It's _way_ better than the vanilla game, the two flaws are that fixed system sizes really screw up the combat and the increased ship agility means that ships can pretty easily jump away.

**Comment by Brick63** [comment]
I don't think you get the point, you have changed your game in a way you believe to be balanced and are trying to make changes to the unchanged game based on the changes. Also the base game is balanced by the fact that it's been years in the making and if you with all your immense "wisdom" think you can make a better game go and make it but don't try to screw up this one.

**Comment by czartrak** [comment]
@accursedCursive OK, so here's where problems arise. Firstly, you seem to ignore anything people say that doesn't coincide with your own view. Secondly, if you increase engine power, of course ships are going to be more agile. It's like saying a car's too fast but you've upgraded it's engines and given it a NOS system

**Comment by Startome** [comment]
I'm going to put this frankly.
The balance issue isn't with ES, it's with your mod.
If your mod doesn't work, make appropriate changes.
But until you actually have contributions to the base game, as opposed to whining because your mod doesn't work, none of us are even mildly interested in anything you have to say.

Your mods the problrm, go fix it.

**Comment by tehhowch** [comment]
@accursedCursive since this doesn't appear to be an issue with the base game, you should probably close this issue.
As far as a solution for your mod, when v0.9.9 comes out you can use a minimal alteration with `add attributes`, i.e.
```
ship "Aerie"
    add attributes
        "mass" X
```
where X is the amount of mass that you want to add to it, e.g. 40, 50, etc. Until v0.9.9 is released, you'll have to settle for reproducing the entire `"attributes"` node in order to get the higher mass that you desire for your plugin.

**Comment by accursedCursive** [comment]
You're all thinking it's an issue with my mod, but it's an issue with the base game that is amplified to noticable levels by my mod.
If Aerie is too agile by a significant margin in my mod, it's too agile by a slim margin in the vanilla game.

**Comment by Amazinite** [comment]
>You're all thinking it's an issue with my mod, but it's an issue with the base game that is amplified to noticable levels by my mod.
If Aerie is too agile by a significant margin in my mod, it's too agile by a slim margin in the vanilla game.

Have you even looked at the numbers I posted? The Aerie is very much middle-of-the-road in terms of the agility of human medium warships.

**Comment by KiLEdEnNis** [comment]
> You're all thinking it's an issue with my mod, but it's an issue with the base game that is amplified to noticable levels by my mod.
If Aerie is too agile by a significant margin in my mod, it's too agile by a slim margin in the vanilla game.

and it never occured to you that such "slim margins" in the base game has neglectable effects? that the base game balances everything compared to said everything?

and did you never even THINK that maybe your mod is the problem?
"the base game has a mini imbalance that's neglectable, it's horribly outbalanced in my plugin, fix it"
that's basically what you're saying

**Comment by Startome** [comment]
That's just it though, I've used these ships many times, and I don't use them now. If they were unbalanced, would I bot use them due to how great they are? In the base game they are balanced well.

Your mod doesn't amplify a problem, it creates a problem where one does not exist.

**Comment by endless-sky** [comment]
Dealing with "issues" like this is exactly why I don't have enough free time to get more worthwhile things (like new Wanderer plot) done.
